### DISCUSSION TITLE
Add oauth ability

### DISCUSSION THREAD

**Discussion by Undertone0809** [fastapi/full-stack-fastapi-template#1292] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
google, github login is a very common requirement, as a basic template, perhaps we can provide oauth login capabilities to build better template services.

If you add google services, add a few two interfaces.


@router.post("/login/google")
async def login_google(request: Request):
    redirect_uri = request.url_for("auth_google")
    return await oauth.google.authorize_redirect(request, redirect_uri)


@router.route('/auth/google')
async def auth_google(request: Request):
    token = await oauth.google.authorize_access_token(request)
    # <=0.15
    # user = await oauth.google.parse_id_token(request, token)
    user = token['userinfo']
    return user
```

### Description

Perhaps what we need to focus on is how to better choose the configuration required by developers. If developers need the configuration of github login, we can easily help users complete this function.

### Operating System

Linux, Windows, macOS, Other

### Operating System Details

Any

### Python Version

3.11

**Comment by alejsdev** [comment, marked as answer]
Hi @Undertone0809

Thanks for your suggestion. This feature is something we plan to include in a future update of this template 😎 🚀
Thank you for your interest!

**Comment by nemanjam** [comment]
Any progress on this? Which example I can use to add login with Github?

**Comment by ffreeez** [comment]
Saw this is still hanging — wanted to push it forward. Before anyone jumps into code, worth hearing what people are thinking.

Library-wise, are we going with authlib, or is there a preference for something else like httpx-oauth, or even a minimal hand-rolled version? Scope: just the common social-login providers, or proper OIDC (ID token validation, discovery, etc.) from the start? Worth bundling RBAC into this too, or keep that as a separate discussion so the PR doesn't balloon? And how far on group/claim sync — create a user on first login and stop there, or sync groups and roles from the IdP on every login? The latter basically pushes this into real SSO territory, which is a different complexity class.

The hardest part is the balance. Template has to stay approachable for newcomers — too many knobs and people bounce — but it also needs enough convenience out of the box that experienced devs don't just rip it out and rewrite. Curious where people think that line should be.

**Comment by nemanjam** [comment]
Meantime I added Github login using Authlib, but also HTTPS cookie instead of localStorage into my fork. Feel free to reuse or use as reference:

https://github.com/nemanjam/full-stack-fastapi-template-nextjs/blob/main/backend/app/api/routes/login.py#L159
